File: lib.rpiws281x/src/test/java/com/externalalarmclock/lib/rpiws281x/image/Rgb2RgbwTest.java
package com.externalalarmclock.lib.rpiws281x.image;

import org.junit.Assert;
import org.junit.Test;

public class Rgb2RgbwTest {
	private final Rgb2Rgbw converter = new Rgb2Rgbw();

	@Test
	public void testBlack() {
		Assert.assertEquals(0, converter.getRgbw(0));
	}
	
	@Test
	public void testWhite() {
		Assert.assertEquals(0xFF000000, converter.getRgbw(0x00FFFFFF));
	}
	@Test
	public void testRedish() {
		Assert.assertEquals(0x19E60000, converter.getRgbw(0x00FF1919));
	}
}

File: lib.rpiws281x/src/test/kotlin/com/externalalarmclock/lib/rpiws281x/image/Rgb2RgbwTest.kt
package com.externalalarmclock.lib.rpiws281x.image

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Rgb2RgbwTest {
    private val converter = Rgb2Rgbw()

    @Test
    fun testBlack() {
        Assertions.assertEquals(0, converter.getRgbw(0).toLong())
    }

    @Test
    fun testWhite() {
        Assertions.assertEquals(-0x1000000, converter.getRgbw(0x00FFFFFF).toLong())
    }

    @Test
    fun testRedish() {
        Assertions.assertEquals(0x19E60000, converter.getRgbw(0x00FF1919).toLong())
    }
}
